docs/interpreter_design_doc.md

```diff
@@ -0,0 +1,100 @@
+# Interpreter Design
```

Comment:
Rename to `interpreter.md` for conciseness?

Reply:
done.